job_definitions/clean_and_migrate_db_dump.yml

```diff
@@ -0,0 +1,101 @@
+- job:
+    name: clean-and-migrate-db-dump
+    display-name: Clean and migrate database dump
+    project-type: pipeline
+    description: Takes the latest production database dump, cleans it, and migrates to target stage. Also Google Drive.
+    concurrent: false
+    triggers:
+      - timed: "0 4 * * 0"
+    parameters:
+      - extended-choice:
+          name: TARGET_STAGE
+          type: radio
+          default-value: staging
+          value: google-drive,preview,staging
+    pipeline:
+      script: |
+        node {
+          currentBuild.displayName = "#${BUILD_NUMBER} - ${TARGET_STAGE}"
+
+            withEnv([
+              "DM_CREDENTIALS_REPO=/var/lib/jenkins/digitalmarketplace-credentials",
+              "CF_HOME=${pwd()}",
+              "STAGE=db-cleanup"
+            ]) {
+
+            try {
+              stage('Prepare') {
+                git url: 'git@github.com:alphagov/digitalmarketplace-aws.git', branch: 'make-db-dumps-clean', credentialsId: 'github_com_and_enterprise'
+                build job: "update-credentials"
+                paas_credentials = sh(script: '$DM_CREDENTIALS_REPO/sops-wrapper -d $DM_CREDENTIALS_REPO/jenkins-vars/paas_credentials_env.enc', returnStdout: true).trim()
+                withEnv(paas_credentials.tokenize("\n")) {
+                  sh('make paas-login')
+                  sh('cf target -s db-cleanup')
+                }
+                env.TARGET_STAGE_ALEMBIC_VERSION = sh(
+                  script: "curl -s https://www.${TARGET_STAGE}.marketplace.team/_status | jq -r '.api_status.db_version'",
+                  returnStdout: true
+                )
+              }
+
+              stage('Deploy db-cleanup app') {
+                sh('make db-cleanup deploy-db-cleanup-app')
+              }
+
+              stage('Import and clean lastest production db-dump') {
+                sh('''
+                  make requirements
+                  make db-cleanup import-and-clean-db-dump
+                ''')
+                env.DUMP_ALEMBIC_VERSION = sh(
+                  script: './scripts/get-alembic-version.sh',
+                  returnStdout: true
+                )
+              }
+
+              stage('Migrate data to target stage and google drive') {
+                sh('GDRIVE_EXPORTDATA_FOLDER_ID="{{ jenkins_gdrive_folder_id }}"  make db-cleanup migrate-cleaned-db-dump')
+              }
+
+              stage('Check if migrations required and run'){
+                if (("${TARGET_STAGE}" == 'preview' || "${TARGET_STAGE}" == 'staging') && (env.DUMP_ALEMBIC_VERSION.toInteger() < env.TARGET_STAGE_ALEMBIC_VERSION.toInteger())) {
```

Comment:
I think it's safe to replace `toInteger` and `<` with `!=`. Migration versions don't have to be numbers, that's just the convention we've been using so far and there's no way for preview or staging to be behind production migration version, so if they're not equal we need to migrate.